src/tests/test_control_flow/Makefile

```diff
@@ -0,0 +1,41 @@
+CC := clang
+LLC := ../../llvm/build/bin/llc
+CPOINTPASS := -Xclang -load -Xclang ../../passes/build/function_passes/libCheckPointPass.so
+UNOPTPASS := -Xclang -load -Xclang ../../passes/build/function_passes/libUnoptimizedCopyPass.so
+LIVEVARPASS := -Xclang -load -Xclang ../../passes/build/function_passes/libLiveVariablesPass.so
+MARKPASS := -Xclang -load -Xclang ./passes/build/basic_block_passes/libMarkUnoptimizedPass.so
+PASSFLAGS := $(CPOINTPASS) $(LIVEVARPASS) $(UNOPTPASS)
+STMAP_CHECKER_DIR := ../../stackmap_checker/
+OBJ_NAMES := utils.o stmap.o jump.o guard.o call_stack_state.o
+OBJS := $(foreach obj, $(OBJ_NAMES), $(STMAP_CHECKER_DIR)$(obj))
+PREFIX := trace
+EXECUTABLES := $(basename $(wildcard trace*.c))
+CLANG_COMPILED := $(foreach obj, $(EXECUTABLES), $(obj)_clang_)
+TARGET_OBJS := $(foreach bin, $(EXECUTABLES), $(bin).o)
+
+.PHONY: all clean stackmap_checker
+
+all: stackmap_checker $(EXECUTABLES)
+
+stackmap_checker:
+	cd $(STMAP_CHECKER_DIR) && $(MAKE)
+
+.SECONDEXPANSION:
```

Comment:
Never seen this before! Why is this needed?

Reply:
It's because I use the `$@.o` variable in the prerequisite of the `$(EXECUTABLES)` rule. Secondary expansion is needed because automatic variables are not normally available in prerequisites. I am escaping `$@.o` (`$$@.o`) and enabling secondary expansion to tell `make` to perform a second round of expansions on the prerequisites of every rule, so that I can actually access `$@`, as described here: http://make.mad-scientist.net/secondary-expansion/